xclim/indices/_agro.py

```diff
@@ -507,6 +508,133 @@ def cool_night_index(
     return cni
 
 
+@declare_units(pr="[precipitation]", evspsblpot="[precipitation]")
+def dryness_index(
+    pr: xarray.DataArray,
+    evspsblpot: xarray.DataArray,
+    lat: xarray.DataArray | str | None = None,
+    wo: str = "200 mm",
+    freq: str = "YS",
+) -> xarray.DataArray:
+    """Dryness Index.
+
+    Mean minimum temperature for September (northern hemisphere) or March (Southern hemisphere).
+    Used in calculating the Géoviticulture Multicriteria Classification System.
+
+    Warnings
+    --------
+    Dryness Index expects CF-Convention conformant potential evapotranspiration (positive up).
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+        Precipitation.
+    evspsblpot : xarray.DataArray
+        Potential evapotranspiration.
```

Comment:
Probably precising `Daily XYZ` could be good? Monthly could also work it seems, although you do a `.resample(freq='MS')`, so probably assuming it's daily?

Reply:
I think we expect all datasets passed into `indices` and `indicators` to be day-length. I can adjust the docstring though.

Reply:
We do have options for other frequencies! See `StandardizedIndexes` in `atmos/precip.py`. It simply needs passing `src_freq=['D', 'MS']` in the base class.

Reply:
Should I change the `Precip` instantiation to `StandardizedIndexes` then?

Reply:
Haha, well it semantically makes no sense, but they are similar enough (same variables, context and input frequencies), so I'm ok with it!

Reply:
> I think we expect all datasets passed into `indices` and `indicators` to be day-length. I can adjust the docstring though.

Yes, at lot of indices have `Daily precipitation` in their docstring, therefore I was wondering if this is something we want to have uniformized. I was not 100% sure, because for some indices it would be ok to pass a monthly precipitation, but I guess this is more the exception than the rule.

Reply:
Regarding `Cool night index`, I see two things:
1. It's based on the following class "tree" (I mean a chain of inheritances, not sure what is the word): `Temp <- Daily <- ResamplingIndicator <- ...`. The class `Daily` defines `src_freq = "D"`, so this is fixed by the class used for this index.  
2. The `allowed_periods = "A"` concerns the resampling frequency, correct? I think this is fine for your needs here. 

Regarding point 1, I guess you can stick with the specific classes, and specify that the source variables are `daily` in the docstring.

Reply:
I could also perform a resampling to daily before running through the calculation? This would allow for finer/smaller time steps.

Reply:
I know that hourly variables are sometimes used for potential_evapotranspiration computations  where a specific method uses hourly data. In this case, I think it would make sense to allow for hourly data, if there's a method related to it. Is this case here, could you use a different method with hourly data? If it's just a matter of convenience, I would suggest to let the users do the transformation `hourly -> daily` themselves. 

The point I was trying to make with monthly data is kind in the opposite direction. In the end, you ask for daily precipitations, but the function is only computing things with the monthly precipitations, correct? So in a sense, if you only have monthly precipitations at hand, you could still compute the `dryness_index` in principle, you just need the function to allow you to do that. Then again, maybe it's a very specific corner case, and users will generally have access daily precips anyways.

Reply:
I have the impression that the approach that reflects more broadly what is done in XClim in general is: Just assume users will input daily precips.

Then, for corner-cases, users have to think about:
* Users with hourly precips should resample before-hand, the docstring  would instruct them to input daily precips.
* Users with monthly precips could use the index anyways, as the monthly resampling `dryness_index` will do nothing. You just have to be aware of this.